src/overview/components/Header.js

```diff
@@ -43,6 +43,14 @@ const Header = ({
             </div>
         </div>
         <div className={styles.links}>
+            <div className={styles.inbox} onClick={props.toggleInbox}>
```

Comment:
You should be able to reduce the modifications needed to overview's `Header` component here (and everything else in the `src/overview` dir). You can move this view to it's own module in `src/notifications/components`, write a simple container to handle fetching/setting the unread count, and then just render the container from this line. 

You would then also not need to modify the overview's `actions` module, you wouldn't need the `unreadNotifCount` state or `toggleInbox` action in the overview container, and you can remove all the new state in overview reducer (move it to the `notifications` dir's reducer).